"panicking in constants is unstable" at build time in 0.7.2

Regression from 0.7.1 to 0.7.2. Getting this at build time:
```
error[E0658]: panicking in constants is unstable
   --> /home/kvark/.cargo/registry/src/github.com-1ecc6299db9ec823/hecs-0.7.2/src/entities.rs:510:21
    |
510 |             None => unreachable!(),
    |                     ^^^^^^^^^^^^^^
    |
    = note: see issue #51999 <https://github.com/rust-lang/rust/issues/51999> for more information
    = note: this error originates in the macro `$crate::panic::panic_2015` (in Nightly builds, run with -Z macro-backtrace for more info)

```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,6 +9,7 @@ repository = "https://github.com/Ralith/hecs"
 readme = "README.md"
 keywords = ["ecs", "entity"]
 categories = ["data-structures", "game-engines", "no-std"]
+rust-version = "1.57"
 
 [package.metadata.docs.rs]
 all-features = true

